### PR TITLE
Add some tests of the catalog modules.

### PR DESCRIPTION
**Pull request by schlafly**
Add tests on the various catalog routines.

### CODE REVIEW

**Comment by schlafly** [comment]
@PaulHuwe , I had to add a # nosec to make bandit happy---I wanted to use urllib to grab some files from galsim's github repository, and bandit doesn't appear to like the urllib repo.  I don't feel too bad about # nosec in this case---the target link is hard coded, so I am not very worried about users inputting something malicious there.  But I don't know what our policy is.

File: romanisim/tests/test_catalog.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    _download_url(cosmos_url, fn)
+    _download_url(cosmos_url.replace('.fits', '_selection.fits'),
+                  fn.replace('.fits', '_selection.fits'))
+    _download_url(cosmos_url.replace('.fits', '_fits.fits'),
+                  fn.replace('.fits', '_fits.fits'))
+    # urllib.request.urlretrieve(cosmos_url, fn)
+    # urllib.request.urlretrieve(cosmos_url.replace('.fits', '_selection.fits'),
+    #                            fn.replace('.fits', '_selection.fits'))
+    # urllib.request.urlretrieve(cosmos_url.replace('.fits', '_fits.fits'),
+    #                            fn.replace('.fits', '_fits.fits'))
```
Comment (PaulHuwe) [inline romanisim/tests/test_catalog.py:38]: Do you want to leave these commented lines here, or should they be cleaned out?

File: romanisim/tests/test_catalog.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    from astropy.io import fits
+    print(fits.getdata(fn).dtype)
```
Comment (PaulHuwe) [inline romanisim/tests/test_catalog.py:40]:
```suggestion
```

File: romanisim/tests/test_catalog.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    _download_url(cosmos_url.replace('.fits', '_selection.fits'),
+                  fn.replace('.fits', '_selection.fits'))
+    _download_url(cosmos_url.replace('.fits', '_fits.fits'),
+                  fn.replace('.fits', '_fits.fits'))
+    # urllib.request.urlretrieve(cosmos_url, fn)
+    # urllib.request.urlretrieve(cosmos_url.replace('.fits', '_selection.fits'),
+    #                            fn.replace('.fits', '_selection.fits'))
+    # urllib.request.urlretrieve(cosmos_url.replace('.fits', '_fits.fits'),
+    #                            fn.replace('.fits', '_fits.fits'))
+    from astropy.io import fits
```
Comment (PaulHuwe) [inline romanisim/tests/test_catalog.py:39]: Move the from to the top?

File: romanisim/tests/test_catalog.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    m = table['type'] == 'PSF'
+    assert np.all(table['n'][m] == -1)
+    assert np.all(table['n'][~m] != -1)
+    assert np.all(table['half_light_radius'][m] == 0)
+    assert np.all(table['half_light_radius'][~m] != 0)
+    assert np.all(table['pa'][m] == 0)
+    assert np.all(table['ba'][m] == 1)
+    assert np.all(table['pa'][~m] != 0)
+    assert np.all(table['ba'][~m] != 0)
+    print(table.dtype.names)
```
Comment (PaulHuwe) [inline romanisim/tests/test_catalog.py:78]:
```suggestion
```

**Review by PaulHuwe — CHANGES_REQUESTED** [review]

**Pull request merged**